Minio/ApiEndpoints/BucketOperations.cs

```diff
@@ -14,170 +14,287 @@
         /// <returns>Task that returns true if exists and user has access</returns>
-        public async Task<bool> BucketExistsAsync(string bucketName, CancellationToken cancellationToken = default(CancellationToken))
+        public async Task<bool> BucketExistsAsync(string bucketName,
+            CancellationToken cancellationToken = default(CancellationToken))
         {
             try
             {
                 var request = await this.CreateRequest(Method.HEAD,
-                                                   bucketName);
-                var response = await this.ExecuteTaskAsync(this.NoErrorHandlers, request, cancellationToken);
+                    bucketName);
+                await this.ExecuteTaskAsync(this.NoErrorHandlers, request, cancellationToken);
             }
             catch (Exception ex)
             {
                 if (ex.GetType() == typeof(BucketNotFoundException))
                 {
                     return false;
                 }
-                throw ex;
+                throw;
             }
             return true;
         }
 
         /// <summary>
-        /// Remove a bucket
+        ///     Remove a bucket
         /// </summary>
         /// <param name="bucketName">Name of bucket to remove</param>
         /// <param name="cancellationToken">Optional cancellation token to cancel the operation</param>
         /// <returns>Task</returns>
-        public async Task RemoveBucketAsync(string bucketName, CancellationToken cancellationToken = default(CancellationToken))
+        public async Task RemoveBucketAsync(string bucketName,
+            CancellationToken cancellationToken = default(CancellationToken))
         {
-            var request = await this.CreateRequest(Method.DELETE, bucketName, resourcePath: null);
-
-            var response = await this.ExecuteTaskAsync(this.NoErrorHandlers, request, cancellationToken);
+            var request = await this.CreateRequest(Method.DELETE, bucketName);
 
+            await this.ExecuteTaskAsync(this.NoErrorHandlers, request, cancellationToken);
         }
 
         /// <summary>
-        /// List all objects non-recursively in a bucket with a given prefix, optionally emulating a directory
+        ///     List all objects non-recursively in a bucket with a given prefix, optionally emulating a directory
         /// </summary>
         /// <param name="bucketName">Bucket to list objects from</param>
         /// <param name="prefix">Filters all objects not beginning with a given prefix</param>
         /// <param name="recursive">Set to false to emulate a directory</param>
         /// <param name="cancellationToken">Optional cancellation token to cancel the operation</param>
         /// <returns>An observable of items that client can subscribe to</returns>
-        public IObservable<Item> ListObjectsAsync(string bucketName, string prefix = null, bool recursive = true, CancellationToken cancellationToken = default(CancellationToken))
```

Comment:
I thought that we should not mix the reactive style with the asynchronous style. We have to add a dependency on the reactive library, which increases the size of the application.